mucommander-core/src/main/java/com/mucommander/ui/quicksearch/QuickSearch.java

```diff
@@ -129,10 +130,17 @@ public boolean matches(String string) {
      * @return true if the given <code>KeyEvent</code> corresponds to a valid quick search input
      */
     protected boolean isValidQuickSearchInput(KeyEvent e) {
-        if((e.getModifiersEx()&(KeyEvent.ALT_DOWN_MASK|KeyEvent.CTRL_DOWN_MASK|KeyEvent.META_DOWN_MASK))!=0)
-            return false;
-
+        int modifier = e.getModifiersEx();
         char keyChar = e.getKeyChar();
+        if ((modifier&(KeyEvent.CTRL_DOWN_MASK|KeyEvent.META_DOWN_MASK)) != 0) {
+            return false;
+        } else if ((modifier & (KeyEvent.ALT_DOWN_MASK)) != 0) {
+            // special case for certain keyboard layouts where diacritics
+            // typed using letter+ALT/OPTION (like Polish Programmer keyboard, the most popular variant in PL).
+            if (!Character.isLetter(keyChar)) {
```

Comment:
I see two possible issues with this:
1. some of the default shortcuts are set to alt+letter, e.g., alt+D and alt+C
2. users may have configured alt+letter shortcuts
while we can change default shortcuts to address (1), I don't see an easy solution to avoid breaking (2)
maybe we should check here if that combination is set as a shortcut for an action first? while I see that it can be important for users that have Polish Programmer keyboards, I believe those are relatively low number of users that can make sure no shortcuts of this form exist

Reply:
lemme think about this :)

Reply:
ad 1 & 2 - pls take look here: https://github.com/mucommander/mucommander/blob/54822d05b53d49301800cfa52a7f857928344c5f/mucommander-core/src/main/java/com/mucommander/ui/main/table/FileTable.java#L1708 - defined shortcuts have already preference

However, here https://github.com/mucommander/mucommander/blob/54822d05b53d49301800cfa52a7f857928344c5f/mucommander-core/src/main/java/com/mucommander/ui/quicklist/item/QuickListDataList.java#L287 there is another use of #isValidQuickSearchInput - and here defined shortcuts are not check, haven't check what QuickListDataList is being used for and whether it matters for this PR.

Reply:
@ahadas convinced? :)

Reply:
quick lists are a bit different because we don't expect operations to be enabled when a quick list is shown but the first part seems to cover the problematic scenario I had in mind